crates/router/src/core/payment_methods/cards.rs

```diff
@@ -84,7 +84,10 @@ pub async fn create_payment_method(
     payment_method_data: Option<Encryption>,
     key_store: &domain::MerchantKeyStore,
 ) -> errors::CustomResult<storage::PaymentMethod, errors::ApiErrorResponse> {
-    db.find_customer_by_customer_id_merchant_id(customer_id, merchant_id, key_store)
+    let merchant_account = db.find_merchant_account_by_merchant_id(merchant_id, key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
```

Comment:
You can pass the merchant account from the previous function. No need of another DB call here

Reply:
sure